### PR TITLE
Fix SAM #1918 and test on Windows 11 and MacOS 15.1.1 with MATLAB

### PR DESCRIPTION
**Pull request by sjanzou**
To test, build SAM and SAM-private if you would like (delete build_windows folder in SAM-private) and run SAM and generate code for some language and check that latest sscapi.h and ssc.dll (or.so or .dylib) is included in the generated code folder.

### CODE REVIEW

**Comment by sjanzou** [comment]
Tested on widows 11
image


Tested on MacOS 15.1.1
image

File: src/codegenerator.cpp [hunk 1]
```diff
@@ -193,12 +193,13 @@ bool CodeGen_Base::PlatformFiles()
...
 	// sscapi.h - switch to copy version for syching issues 5/30/17
 	// assumes in runtime folder for all builds.
-	f1 = SamApp::GetAppPath() + "/sscapi.h";
+	f1 = SamApp::GetRuntimePath() + "/sscapi.h";
 	if (wxFileExists(f1)) {
 		f2 = m_folder + "/sscapi.h";
 		wxCopyFile(f1,f2);
 	}
-	else {
+	//else {
```
Comment (mjprilliman) [inline src/codegenerator.cpp:201]: Is this going to be needed in the future? If not, can we remove the commented part?

File: src/codegenerator.cpp [hunk 2]
```diff
@@ -193,12 +193,13 @@ bool CodeGen_Base::PlatformFiles()
...
+		/* skip fallback per SAM issue 1918
```
Comment (mjprilliman) [inline src/codegenerator.cpp:202]: Same question

Comment (sjanzou) [reply src/codegenerator.cpp:202]: Removed commented code as requested - was initially left as a fallback position but more confusing and out of date. Thanks for the future maintenance request!

**Pull request merged**